starboard/shared/starboard/media/codec_util.cc

```diff
@@ -107,13 +108,101 @@ SbMediaAudioCodec GetAudioCodecFromString(const char* codec,
     return kSbMediaAudioCodecPcm;
   }
 #if SB_API_VERSION >= 15
-  if (strcmp(codec, "iamf") == 0 || strncmp(codec, "iamf.", 5) == 0) {
+  if (strcmp(codec, "iamf") == 0 || IsIamfMimeType(codec)) {
     return kSbMediaAudioCodecIamf;
   }
 #endif  // SB_API_VERSION >= 15
   return kSbMediaAudioCodecNone;
 }
 
+#if SB_API_VERSION >= 15
+bool IsIamfMimeType(std::string mime_type) {
+  // Reference: Immersive Audio Model and Formats;
+  //            v1.0.0
+  //            6.3. Codecs Parameter String
+  // (https://aomediacodec.github.io/iamf/v1.0.0-errata.html#codecsparameter)
+  if (mime_type.find("iamf") == std::string::npos) {
+    return false;
+  }
+
+  // 4   FOURCC string "iamf".
+  // +1  delimiting period.
+  // +3  primary_profile as 3 digit string.
+  // +1  delimiting period.
+  // +3  additional_profile as 3 digit string.
+  // +1  delimiting period.
+  // +9  The remaining string is one of "Opus", "mp4a.40.2", "fLaC", or "ipcm".
+  constexpr int kMaxIamfCodecIdLength = 22;
+
+  if (mime_type.size() > kMaxIamfCodecIdLength) {
+    return false;
+  }
+
+  std::vector<std::string> vec = SplitString(mime_type, '.');
+  if (vec.size() != 4 && vec.size() != 6) {
+    return false;
+  }
+
+  if (vec[0] != "iamf") {
+    return false;
+  }
+
+  // The primary profile string should be three digits, and should be between 0
+  // and 255 inclusive.
+  int primary_profile;
+  std::stringstream stream(vec[1]);
+  stream >> primary_profile;
+  if (stream.fail() || vec[1].size() != 3 || primary_profile > 255) {
+    return false;
+  }
+
+  // The additional profile string should be three digits, and should be between
+  // 0 and 255 inclusive.
+  stream = std::stringstream(vec[2]);
+  int additional_profile;
+  stream >> additional_profile;
+  if (stream.fail() || vec[2].size() != 3 || additional_profile > 255) {
+    return false;
+  }
+
+  // The codec string should be one of "Opus", "mp4a", "fLaC", or "ipcm".
+  std::string codec = vec[3];
```

Comment:
Maybe consider to refactor the following code to switch-case, so it is more readable.

For example:
```
switch(codec) {
  case "Opus":
  case "fLaC":
  case "ipcm":
    return true;
  default:
    // handle "mp4a.40.2" and check its format here
    // return true if it meets
}

return false;
```

Reply:
Unfortunately switch statements can't work with strings, only values resolving to int.